src/code/templates/clutch-game.js

```diff
@@ -495,45 +498,23 @@ export default class ClutchGame extends Component {
     return [shuffle(fatherPool), shuffle(motherPool)];
   }
 
-  static authoredDrakesToDrakeArray = function(authoredChallenge, trial) {
-    const mother = new BioLogica.Organism(BioLogica.Species.Drake,
-                                          authoredChallenge.mother.alleles,
-                                          authoredChallenge.mother.sex),
-          father = new BioLogica.Organism(BioLogica.Species.Drake,
-                                          authoredChallenge.father.alleles,
-                                          authoredChallenge.father.sex),
-          // authored specs may be incomplete; these are complete specs
-          motherSpec = { alleles: mother.getAlleleString(), sex: mother.sex },
-          fatherSpec = { alleles: father.getAlleleString(), sex: father.sex };
-
-    // already generated drakes
-    if (trial > 0)
-      return authoredDrakes;
-
-    // challengeType === 'match-target'
-    targetDrakes = authoredChallenge.targetDrakes;
-    const targetDrakeCount = targetDrakes.length;
-
-    function childDrakesContain(alleles) {
-      for (let i = 3; i < authoredDrakes.length; ++i) {
-        if (authoredDrakes[i].alleles === alleles)
-          return true;
-      }
-      return false;
-    }
-
-    authoredDrakes = [motherSpec, fatherSpec];
-    for (let i = 0; i < targetDrakeCount; ++i) {
-      let child, childAlleles;
-      do {
-        child = BioLogica.breed(mother, father, false);
-        childAlleles = child.getAlleleString();
-        // don't generate the same set of alleles twice
-      } while (childDrakesContain(childAlleles));
-
-      authoredDrakes.push({ alleles: childAlleles, sex: child.sex });
+  static authoredDrakesToDrakeArray = function(authoredChallenge, trial, unscrambledTrialNumber) {
+    const changeableDrake = new BioLogica.Organism(BioLogica.Species.Drake,
+                                                   authoredChallenge.changeableDrakes[trial].alleles,
+                                                   BioLogica.FEMALE),
+          staticDrake = new BioLogica.Organism(BioLogica.Species.Drake,
+                                               authoredChallenge.staticDrakes[trial].alleles,
+                                               BioLogica.FEMALE);
+    changeableDrakeType = authoredChallenge.changeableDrakeType[unscrambledTrialNumber];
+    let fatherSpec, motherSpec;
+    if (changeableDrakeType === "mother") {
+      motherSpec = { alleles: changeableDrake.getAlleleString(), sex: BioLogica.FEMALE };
+      fatherSpec = { alleles: staticDrake.getAlleleString(), sex: BioLogica.MALE };
+    } else {
+      motherSpec = { alleles: staticDrake.getAlleleString(), sex: BioLogica.FEMALE };
+      fatherSpec = { alleles: changeableDrake.getAlleleString(), sex: BioLogica.MALE };
     }
-    return authoredDrakes;
+    return [motherSpec, fatherSpec, authoredChallenge.targetDrakes[trial]];
```

Comment:
I like this suggestion! Much cleaner than the changes I made here, and give what you pointed out during the demo yesterday (i.e. my levels aren't as flexible as I thought), I'm going to make this change